airflow_dbt_python/hooks/dbt.py

```diff
@@ -454,11 +443,14 @@ def get_dbt_target_from_connection(
             )
             return None
 
-        details = self.get_dbt_details_from_connection(conn)
+        db_hook_class = self.get_db_hook_class(conn)
+
+        details = db_hook_class.get_dbt_details_from_connection(conn)
 
         return {conn_id: details}
 
-    def get_dbt_details_from_connection(self, conn: Connection) -> dict[str, Any]:
+    @classmethod
+    def get_dbt_details_from_connection(cls, conn: Connection) -> dict[str, Any]:
```

Comment:
I am not following why this has to be a classmethod now. This change also means we need the additional `get_db_hook_class` to resolve which class we are calling from, which just seems confusing as we go back and forth in the MRO. 

Couldn't we just have overriden `get_dbt_details_from_connection` on each concrete `DbtHook` subclass? Since we are having to do that anyways.

Reply:
Closing comment as no longer relevant.